src/opensearch_sdk_py/transport/stream_output.py

```diff
@@ -35,6 +35,23 @@ def write_v_int(self, i: int) -> int:
                 break
         return self.write(result)
 
+    @classmethod
+    def v_int_size(self, i: int) -> int:
+        if i < 0x80:
```

Comment:
This works, but in 99.999999% of cases we'll be looking at OpenSearch versions which are 4 bytes, so I'd suggest a shortcut before this where if `i > 0xffffff` we return 4.

Reply:
I like this: https://github.com/opensearch-project/opensearch-sdk-py/pull/14